cgc/coclustering_dask.py

```diff
@@ -74,31 +68,31 @@ def coclustering(Z, nclusters_row, nclusters_col, errobj, niters, epsilon,
     s = 0
     converged = False
 
-    Gavg = Z.mean()
-
     while (not converged) & (s < niters):
         logger.debug(f'Iteration # {s} ..')
         # Calculate cluster based averages
         # nel_clusters is a matrix with the number of elements per co-cluster
         # originally computed as:  da.dot(da.dot(R.T, da.ones((m, n))), C)
         nel_row_clusters = da.bincount(row_clusters, minlength=nclusters_row)
         nel_col_clusters = da.bincount(col_clusters, minlength=nclusters_col)
+        row_cluster_labels = nel_row_clusters.nonzero()[0].compute()
+        col_cluster_labels = nel_col_clusters.nonzero()[0].compute()
         logger.debug('num of populated clusters: row {}, col {}'.format(
-                        da.sum(nel_row_clusters > 0).compute(),
-                        da.sum(nel_col_clusters > 0).compute()))
-        R = _setup_cluster_matrix(nclusters_row, row_clusters)
-        C = _setup_cluster_matrix(nclusters_col, col_clusters)
-        nel_clusters = da.outer(nel_row_clusters, nel_col_clusters)
-        CoCavg = (da.matmul(da.matmul(R.T, Z), C) + Gavg * epsilon) / \
```

Comment:
I have a very very minor doubt about removing `Gavg`. I guess it should be fine to totally remove it since it only functions with epsilon. But maybe we should just double-check with Emma?